readme.md

```diff
@@ -385,90 +507,53 @@ interface SubmitTerms {
-    legitimations?: ICredential[]
+     *  @link https://github.com/KILTprotocol/kilt-extension-api/blob/4c0c2f93958ab72b59b72057a6e9b6aedb5fccef/src/types/Quote.ts#L30 */
+    quote?: IQuote
+    proof: VCDataIntegrity
 }
 ```
 
 
-#### 2. Extension requests credential
+#### 2.a Extension requests credential
 
 The extension MUST only send the request with active consent of the user.
 This is the first step where the user’s DID is revealed to the dApp.
 
-The previous message in the flow - `'submit-terms'` - contains the `claim` with an optional `owner` field containing a DID URI.
-This `owner` value being provided means that the attester is willing to issue the credential for this specific DID.
-Note that legacy attesters provide bogus DIDs in this field, this is likely the case if the extension does not control this DID.
-If so, the extension SHOULD proceed as if `owner` was not provided.
+The previous message in the flow - `'submit-terms'` - contains the `claim` with an optional `subject` field containing a DID URI.
+This `subject` value being provided means that the attester is willing to issue the credential for this specific DID.
+If the `'submit-terms'` message included an unknown DID or none at all as `subject`, the extension MUST ask the user to choose the DID for which the credential will be issued.
+Otherwise, the extension SHOULD NOT offer the choice, but still MUST get the user’s consent to use this DID.
+
+The extension MUST generate the `salt` values according to the [KiltAttestationProofV1 specification](https://github.com/KILTprotocol/spec-KiltCredentialV1/blob/main/ProofTypes/KiltAttestationProofV1.md#salt) and provide them in the `request-attestation` message.
+The extension MUST provide the `delegationId` if the attester provided one in the previous step.
 
-If the `'submit-terms'` message included an unknown DID or none at all as `owner`, the extension MUST ask the user
-to choose the DID for which the credential will be issued. Otherwise, the extension SHOULD NOT offer the choice,
-but still MUST get the user’s consent to use this DID.
+The extension used a temporary DID for the communication channel.
+Since this DID is only used for a single interaction, it is not possible for the attester to verify the identity of the claimer until this step.
+In order for the attester to hold a proof that the claimer agreed to the terms, the claimer MUST sign the `RequestAttestation` object using the DID specified in the credential subject (field `claim.credentialSubject.id`).
+The attester SHOULD store the signed `RequestAttestation` object in case it is needed for later dispute resolution.
 
-The chosen or confirmed DID URI will be submitted as the `owner` field of the `claim` in the `'request-attestation'` message.
+
+The chosen or confirmed DID URI will be submitted as the `subject` field of the `claim` in the `'request-attestation'` message.
```

Comment:
again, I don't think there is a `subject`

Reply:
[fixed in this commit](315037d22e3539d6620f4b6d2c616fd200d0d068) hope the JSON pointer is ok? Not sure it should be relative to the message object or the content of the message. So `/claim/credentialSubject/id` or `/body/content/claim/credentialSubject/id`

Reply:
I think people will understand